code/modules/food_and_drink/snacks.dm

```diff
@@ -2206,6 +2206,45 @@ var/list/valid_jellybean_reagents = childrentypesof(/datum/reagent)
 
 		boutput(M, "[color][phrase]! That tasted like [flavor]...</span>")
 
+/obj/item/reagent_containers/food/snacks/candy/everyflavor/some
+	name = "\improper Farty Snott's Some Flavour Bean"
+	desc = "A home-grown organic version of the original favorite halloween sweet!"
+
+	New()
+		..()
```

Comment:
Will this not just do the reagent filling for regular everyflavor beans?

Reply:
ah fuck
youre right, im not sure if theres a way to call my grandparents hmmmm

Reply:
@pali6 pls offer sagely advice, should i just make a new someflavor bean object that doesn't inherit from this?

Reply:
Hi, not-pali here!

I would look to alter the `/obj/item/reagent_containers/food/snacks/candy/everyflavor` to `/obj/item/reagent_containers/food/snacks/candy/jellybean`, which has most of the same behavior as the `everyflavor` one currently, but give it a `var/random_flavor_list` that is your whitelist of reagents. You can then have `/obj/item/reagent_containers/food/snacks/candy/jellybean/everyflavor` which is the exact same except for its name and setting `random_flavor_list = all_functional_reagent_ids`.

Reply:
Oh, and you'd need to change references to `.../everyflavor` to the new one. The reason I'd do it this way (rather than have the default behavior be all reagent IDs and the `/someflavor` one shorten the list) is basically that the default one is the "safe" one and the specialized one is the more dangerous one.

Reply:
@mordent-goonstation that's a really good shout actually, i'll make those changes at some point over the weekend :)

Reply:
Submitted a commit to resolve this issue